as/itests/org.jboss.tools.as.ui.bot.itests/src/org/jboss/tools/as/ui/bot/itests/parametized/server/ServerAdaptersTest.java

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * JBIDE-26853
+ * Tests creation of recent server adapters
+ * 
+ * @author jkopriva@redhat.com
+ *
+ */
+@RunWith(RedDeerSuite.class)
+@JRE(cleanup=true)
+@UseParametersRunnerFactory(ParameterizedRequirementsRunnerFactory.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ServerAdaptersTest extends AbstractTest {
+
+	private final Logger LOGGER = Logger.getLogger(this.getClass());
+	
+	public static final String WILDFLY_FAMILY = "JBoss Community";
+	public static final String EAP_FAMILY = "Red Hat JBoss Middleware";
+
+	@Parameters(name = "{0}")
+	public static ArrayList<String> data() {
+		ArrayList<String> list = new ArrayList<String>();
+		list.add("WildFly 17");
+		list.add("WildFly 18");
+		list.add("Red Hat JBoss Enterprise Application Platform 7.0");
+		list.add("Red Hat JBoss Enterprise Application Platform 7.1");
+		list.add("Red Hat JBoss Enterprise Application Platform 7.2");
+		list.add("Red Hat JBoss Enterprise Application Platform 7.3 (Tech Preview)");
+
+		return list;
+	}
+
+	@BeforeClass
+	public static void prepareWorkspace() {
+		deleteRuntimes();
+	}
+
+	private String server;
+
+	public ServerAdaptersTest(String server) {
+		this.server = server;
+	}
+
+	@Test
+	public void setupLocalServerAdapter() {
+		NewServerWizard serverW = new NewServerWizard();
+		try {
+			serverW.open();
+
+			NewServerWizardPage sp = new NewServerWizardPage(serverW);
+
+			sp.selectType(getFamily(server), server);
+
+			serverW.next();
+
+			NewServerAdapterPage ap = new NewServerAdapterPage(serverW);
+			ap.setRuntime(null);
+			ap.checkErrors();
+
+			serverW.next();
+
+			setupRuntime(serverW);
+
+			serverW.finish();
+		} catch (AssertionError | RuntimeException e) {
+			try {
+				serverW.cancel();
+			} catch (RedDeerException ex) {
+				LOGGER.error("Cannot close server wizard!", ex);
+			}
+			throw e;
+		} 
+	}
+
+	private String getFamily(String server) {
+		if (server.contains("WildFly")) {
+			return WILDFLY_FAMILY;
+		} else {
+			return EAP_FAMILY;
+		}
+	}
+
+	protected void setupRuntime(NewServerWizard wizard) {
+		JBossRuntimeWizardPage rp = new JBossRuntimeWizardPage(wizard);
+		rp.setRuntimeName(this.server + " Runtime");
+		rp.setRuntimeDir(getDownloadPath().getAbsolutePath());
+	}
+
+	protected File getDownloadPath() {
+		return new File(getServerHome(this.server));
+	}
+
+	private String getServerHome(String server) {
+		String homeFlag;
+		if (server.contains("WildFly")) {
+			String version = server.split(" ")[1];
+			homeFlag = Arrays.stream(PomServerConstants.getJBossHomeFlags()).filter(x -> x.contains(version))
+					.findFirst().orElse(null);
+		} else {
+			String version = server.split(" ")[6];
+			homeFlag = Arrays.stream(PomServerConstants.getJBossHomeFlags()).filter(x -> (x.contains(version) && x.contains("eap")))
+					.findFirst().orElse(null);
+		}
+		return System.getProperty(homeFlag);
+	}
+
+	@After
+	public void deleteWorkspace() {
+		deleteRuntimes();
```

Comment:
I would delete this. If you add another test this will probably create an test error as JRE will be missing afterwards...

Reply:
This is related to server runtimes, not JREs.

Reply:
In that case...